tests/pytest_tests/system_tests/test_importers/test_wandb/conftest.py

```diff
@@ -0,0 +1,300 @@
+        )
+        command = ["docker", "volume", "rm", config.wandb_server_settings.volume]
+        subprocess.run(command, check=True)
+
+        if config.getoption("--wandb-second-server"):
+            print(
+                f"Cleaning up wandb server volume2 ({config.wandb_server_settings2.volume}) ..."
+            )
+            command = ["docker", "volume", "rm", config.wandb_server_settings2.volume]
+            subprocess.run(command, check=True)
+
+
+def determine_scope(fixture_name, config):
+    return config.getoption("--user-scope")
+
+
+@pytest.fixture(scope="session")
+def fixture_fn2(request, fixture_fn_factory):
+    yield from fixture_fn_factory(request.config.wandb_server_settings2)
+
+
+@pytest.fixture(scope=determine_scope)
+def user2(request, user_factory, fixture_fn2):
+    yield from user_factory(fixture_fn2, request.config.wandb_server_settings2)
+
+
+@pytest.fixture
+def server_src(user):
+    n_experiments = 2
+    n_steps = 50
+    n_metrics = 3
+    n_reports = 2
+    project_name = "test"
+
+    for _ in range(n_experiments):
+        run = wandb.init(entity=user, project=project_name)
+
+        # log metrics
+        data = generate_random_data(n_steps, n_metrics)
+        for i in range(n_steps):
+            metrics = {k: v[i] for k, v in data.items()}
+            run.log(metrics)
+
+        # log tables
+        run.log(
+            {
+                "df": create_random_dataframe(),
+                "img": create_random_image(),
+                # "vid": create_random_video(),  # path error matplotlib
+                "audio": create_random_audio(),
+                "pc": create_random_point_cloud(),
+                "html": create_random_html(),
+                "plotly_fig": create_random_plotly(),
+                "mol": create_random_molecule(),
+            }
+        )
+
+        # log artifacts
+        for _ in range(2):
+            art = make_artifact("logged_art")
+            run.log_artifact(art)
+
+        art2 = make_artifact("used_art")
+        run.use_artifact(art2)
+        run.finish()
+
+        # TODO: We should be testing for gaps in artifact sequences (e.g. if an artifact was deleted).
+        # In manual tests it does work, but it seems to misbehave in the testcontainer, so commenting
+        # this out for now.
+        # delete the middle artifact in sequence to test gap handling
+        # api = wandb.Api()
+        # art_type = api.artifact_type("logged_art", project_name)
+        # for collection in art_type.collections():
+        #     for art in collection.artifacts():
+        #         v = int(art.version[1:])
+        #         if v == 1:
+        #             art.delete(delete_aliases=True)
+
+    # create reports
+    for _ in range(n_reports):
+        wr.Report(project=project_name, blocks=[wr.H1("blah")]).save()
+
+
+def generate_random_data(n: int, n_metrics: int) -> list:
```

Comment:
Does the randomness matter for the test? If there's a bug that only happens for some combinations data, then at best the test is flaky. It would be better to use the simplest possible data for which the test still exercises the relevant codepaths.

Reply:
Randomness does not matter, except maybe for fuzzing

We can make the data not random

Reply:
It's best to avoid calling `np.random.seed()` and `random.seed()`. The idiomatic NumPy way is to create a `RandomState`:

```py
def generate_random_data(n: int, n_metrics: int) -> list:
    random = np.random.RandomState(seed=1337)

    # Then use the `random` variable rather than `np.random`.
    #   np.random.uniform() => random.uniform()
    #   random.random() => random.uniform()
    #   random.choice() => random.choice()  (uses the RandomState now)
    ...
```